app/controllers/idv/cancellations_controller.rb

```diff
@@ -0,0 +1,24 @@
+module Idv
+  class CancellationsController < ApplicationController
+    include IdvSession
+
+    before_action :confirm_two_factor_authenticated
+    before_action :confirm_idv_needed
+
+    def new
+      @presenter = CancellationPresenter.new(step: current_step, view_context: view_context)
+    end
+
+    def destroy
+      @presenter = CancellationConfirmationPresenter.new
+      idv_session = user_session[:idv]
+      idv_session&.clear
+    end
+
+    private
+
+    def current_step
+      params.require(:step).to_sym
```

Comment:
We know they should be on that step because they clicked the cancel link on that step, landing them here. Also, working now on getting rid of the current_step in favor of the referrer.